rskj-core/src/test/java/co/rsk/net/discovery/MessageDecoderTest.java

```diff
@@ -0,0 +1,152 @@
+package co.rsk.net.discovery;
+
+import co.rsk.net.discovery.message.*;
+import org.bouncycastle.util.encoders.Hex;
+import org.ethereum.crypto.ECKey;
+import org.junit.Assert;
+
+import java.util.ArrayList;
+import java.util.UUID;
+import org.junit.Test;
+
+
+public class MessageDecoderTest {
+
+    private static final String KEY_1 = "bd1d20e480dfb1c9c07ba0bc8cf9052f89923d38b5128c5dbfc18d4eea38261f";
+    private static final int NETWORK_ID = 1;
+
+    @Test(expected = PeerDiscoveryException.class)
+    public void testMDCCheckFail() {
+        byte[] wire = Hex.decode("00000000000000000000000000000000000000000000000000000000000000000000000000000000" +
+                "000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000" +
+                "000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000" +
+                "000000000000000000000000000000000000000000000000000000000000000000");
+        MessageDecoder.decode(wire);
+        Assert.fail();
+    }
+
+    @Test(expected = PeerDiscoveryException.class)
+    public void testLengthFail() {
+        MessageDecoder.decode(new byte[] {11});
+        Assert.fail();
+    }
+
+    @Test(expected = PeerDiscoveryException.class)
+    public void testDataSizeNeighborsMessage() {
+        byte[] wire = Hex.decode("64161b75a22291a416f2b3ad0c9e69311bda74d0fd8b168c018228fa1c9ecd33352771ac6b300bd2" +
```

Comment:
Wouldn't be possible to create this binaries programmatically? it looks like it would be pretty difficult to maintain this tests in the future written this way